### PR TITLE
Add PostGIS Timestamp support (Fix 678)

### LINKED ISSUE #678: [FEATURE]postgres provider: add support for timestamp types
Original commit: https://github.com/qgis/QGIS/commit/f6e4e75cd6257a2fa6a59beb9892d7517bbbd39d by jef-n

Unfortunately this lazy coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by yjacolin**
Fix #678

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 1]
```diff
@@ -587,6 +587,8 @@ trouble loading a PostgreSQL table into QGIS, the information below may
 help you understand any QGIS messages and give you direction on changing
 the PostgreSQL table or view definition to allow QGIS to load it.
 
+**Primary key**
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Just wonder if bolding better suits than creating real subsection...

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 2]
```diff
@@ -596,6 +598,8 @@ these items, the oid column will be used instead. Performance will be
 improved if the column is indexed (note that primary keys are automatically
 indexed in PostgreSQL).
 
+**View**
+
 If the PostgreSQL layer is a view, the same requirement exists, but views
 do not have primary keys or columns with unique constraints on them. You have to
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: do not **always** have

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 3]
```diff
@@ -604,27 +608,41 @@ will not load the layer. If this occurs, the solution is to alter the view so
 that it does include a suitable column (a type of integer and either a primary
 key or with a unique constraint, preferably indexed).
 
+**Define primary key**
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Would that be possible to merge this part with the primary key section?

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 4]
```diff
@@ -604,27 +608,41 @@ will not load the layer. If this occurs, the solution is to alter the view so
...
+
 QGIS offers a checkbox **Select at id** that is activated by default. This
 option gets the ids without the attributes which is faster in most cases. It
 can make sense to disable this option when you use expensive views.
 
-.. tip:: **Backup of PostGIS database with layers saved by QGIS**
+**QGIS layer_style table and backup**
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: QGIS layer_style table and **database** backup
Could you add an anchor here for me please? I would like to be able to link this part directly from the style section

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 5]
```diff
@@ -604,27 +608,41 @@ will not load the layer. If this occurs, the solution is to alter the view so
...
+
 QGIS allows to filter features already on server side. Check the
 |checkbox| :guilabel:`Execute expressions on postgres server-side if
 possible (Experimental)` checkbox to do so. Only supported expressions will be
 sent to the database. Expressions using unsupported operators or functions will
 gracefully fallback to local evaluation.
 
+**Support of PostgreSQL data types**
+
+Most of common data types are supported by the QGIS PostgreSQL provider: integer, float,
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: ~~QGIS~~ PostgreSQL

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 6]
```diff
@@ -604,27 +608,41 @@ will not load the layer. If this occurs, the solution is to alter the view so
...
+varchar, geometry and timestamp.
+
+Array data types are not supported.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: This is true (to ease backport of the commit in 2.18). Then I guess you need to review it to also fix #1406

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 3 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: well, this is not always focus on views, except the last sentence.

Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: I splited the paragraph in two part. Let's see if this is better (wait for the push ;) )

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 4 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: `qgis_layer_style_dbbackup` not find a shorter one, this is the third day in the hackfest :)

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 6 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: I will make another PR for #1406. thanks to point to this.

File: source/docs/user_manual/working_with_vector/supported_data.rst (hunk 4 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: `layer_styles_backup` was enough (it's just for writers - as long as it's unique and explicite, no need to complicate :smile: )

Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: simplified!

**Pull request merged**